### PR TITLE
Added kernel upgrade workflow documentation

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
@ddysher PTAL
@multilinear FYI re:  #5044

**Comment by erictune** [comment]
Depends on #4585

File: docs/cluster_management.md [hunk 1]
```diff
@@ -31,3 +31,25 @@ You can use the kube-version-change utility to convert config files between diff
 $ hack/build-go.sh cmd/kube-version-change
 $ _output/go/bin/kube-version-change -i myPod.v1beta1.yaml -o myPod.v1beta3.yaml
 ```
+
+### Upgrading the kernel on a node
```
Comment (eparis) [inline docs/cluster_management.md]: What does this have to do with upgrading a kernel?  (Yes it's a good example, but updating libc would have the same issues)

This is about preparing a node for cluster removal, either temporary or permanent...   I'll try to think of a better title, but saying this is only about kernel isn't good...

Comment (dchen1107) [inline docs/cluster_management.md]: +1. Upgrading docker daemon requires downtime, and all running containers are killed today too.

Comment (erictune) [inline docs/cluster_management.md]: The containers have their own libc's, so, in principle, you could upgrade the node libc, restart all the node daemons, and not affect any running containers, and not have any open file handles on the old libc.

Comment (ddysher) [inline docs/cluster_management.md]: Does docker daemon use libc?  It seems we cant avoid downtime in the libc case.

How about `Maintenance event on a node`?

File: docs/cluster_management.md [hunk 2]
```diff
@@ -31,3 +31,25 @@ You can use the kube-version-change utility to convert config files between diff
...
+  1. Mark the node to be rebooted as unschedulable:
+    `kubectl update nodes $NODENAME --patch='{"apiVersion": "v1beta1", "unschedulable": true}'. 
+    This keeps new pods from landing on the node while you are trying to get them off.
+  1. Get the pods off the machine, via any of the following strategies:
+    1. wait for finite-duration pods to complete
+    1. for pods with a replication controller, delete the pod with `kubectl delete pods $PODNAME`
+    1. for pods which are not replicated, bring up a new copy of the pod, and redirect clients to it.
+  1. Work on the node
+  1. Make the node schedulable again:
+    `kubectl update nodes $NODENAME --patch='{"apiVersion": "v1beta1", "unschedulable": false}'.
```
Comment (ddysher) [inline docs/cluster_management.md]: missed an ending `

File: docs/cluster_management.md [hunk 3]
```diff
@@ -31,3 +31,25 @@ You can use the kube-version-change utility to convert config files between diff
...
+     Or, if you deleted the VM instance and created a new one, and are using `--sync_nodes=true` on the apiserver
+     (the default), then a new schedulable node resource will be created automatically when you create a new
+     VM instance.  See [docs/node.md].
```
Comment (ddysher) [inline docs/cluster_management.md]: s/[docs/node.md]/[Node](docs/node.md)

File: docs/cluster_management.md (hunk 1 above)
Comment (erictune) [inline docs/cluster_management.md]: Called it "Upgrading the system software on a Node"

Comment (erictune) [inline docs/cluster_management.md]: Okay, called it "Maintenance on a Node"

File: docs/cluster_management.md (hunk 2 above)
Comment (erictune) [inline docs/cluster_management.md]: fixed.

**Comment by erictune** [comment]
PTAL

**Pull request merged**